### DISCUSSION TITLE
being able to set the image filename

### DISCUSSION THREAD

**Discussion by laurenskling** [gatsbyjs/gatsby#35866] [Ideas / Feature Requests]
My images are sourced from Strapi, which adds a hash to all images. (e.g.: image_e2cba.jpg)
my clients wants to have full control over the image names for SEO purposes. I'd be great if I can programmatically change the filename `gatsby-plugin-image` outputs. Strapi has a `filename` field in the UI. But there's no way for me to use that as the actual filename in the gatsby output (and Strapi doesn't let me rename files).

So being able to do something like:

```
<GatsbyImage
  image={getImage(image.localFile)}
  alt={image.alternativeText}
  filename={image.name}
/>
```

**Comment by laurenskling** [comment]
I just realised my thinking was way too easy. This is too dynamic, how would the sharp images get their name in building to the static folder.. So it should be in the Graphql side of things, but how would you put a dynamic value in your query :/

**Comment by bre7** [comment]
Or maybe allow each image to be renamed only once ? Came across https://github.com/gatsbyjs/gatsby/issues/15023#issuecomment-518153881 as a possible solution

~~Repo available in archive.org https://web.archive.org/web/20201207082847/https://github.com/sidharthachatterjee/gatsby-custom-filenames~~

**Comment by laurenskling** [comment]
The only solution I can see right now using the `gatsbyImageData` function in GraphQL, but still, how would that receive the data you'd want to use for a filename. For example, I want to put the page title as part of the image URL. How could I tell `gatsbyImageData` about that piece of data?
